src/Sulu/Bundle/MediaBundle/Api/Collection.php

```diff
@@ -118,38 +199,51 @@ public function getId()
      */
     public function getMediaNumber()
     {
-        $mediaCount = 0;
-
-        /** @var Entity $child */
-        foreach ($this->entity->getChildren() as $child) {
-            $mediaCount += count($child->getMedia());
-        }
-
-        // set media count
-        $mediaCount += count($this->entity->getMedia());
-
-        return $mediaCount;
+        return count($this->entity->getMedia());
     }
 
     /**
-     * @param Entity $parent
+     * @param Collection $parent
      * @return $this
      */
     public function setParent($parent)
     {
-        $this->entity->setParent($parent);
+        if ($parent !== null) {
+            $this->entity->setParent($parent->getEntity());
+        } else {
+            $this->entity->setParent(null);
+        }
+
+        // FIXME cache for parent because of the preview images which need a service to create them
+        $this->parent = $parent;
 
         return $this;
     }
 
     /**
-     * @VirtualProperty
-     * @SerializedName("parent")
-     * @return int|null
+     * @return Collection
      */
     public function getParent()
     {
-        return ($parent = $this->entity->getParent()) ? $parent->getId() : null;
+        // FIXME cache for parent because of the preview images which need a service to create them
+        //       generate it on the fly
```

Comment:
Can you please check this file for this comment once more? Here you write `cache for parent ...` and in the children method you write `remove cache for children`, which is definitely not the same ;-)